arangod/Agency/Supervision.cpp

```diff
@@ -1009,8 +1038,8 @@ void Supervision::run() {
 // Guarded by caller
 bool Supervision::isShuttingDown() {
   _lock.assertLockedByCurrentThread();
-  if (_snapshot.has("Shutdown")) {
-    return _snapshot.hasAsBool("Shutdown").first;
+  if ( _snapshot && _snapshot->has("Shutdown")) {
```

Comment:
Why do we check in some places if `_snapshot` is non-nullptr bui not in all places?